### PR TITLE
Resource quota webhook providers - AWS, GCP, Azure, KubeVirt, vSphere, Openstack

### LINKED ISSUE #9626: Add resource quota calculation per provider in machine webhook
For each provider implement getting requested node size from a machine

### PR DESCRIPTION
**Pull request by lsviben**
**What does this PR do / Why do we need it**:
adds logic for getting providers(AWS, GCP, Azure, KubeVirt, vSphere, Openstack) machine resource details to the resource quota machine webhook.

For now getting the ResourceQuota for the project is still hardcoded (because ResourceQuota is not implemented yet), so the resource calculation is skipped until the full implementation.

**Does this PR close any issues?**:
Fixes #9626

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/ee/validation/machine/resourcequota.go [hunk 1]
```diff
@@ -51,7 +53,37 @@ func ValidateQuota(ctx context.Context, log *zap.SugaredLogger, seedClient ctrlr
...
+		if err != nil {
+			return fmt.Errorf("error getting azure resource requirements: %w", err)
+		}
+	case types.CloudProviderKubeVirt:
+		quotaReq, err = getKubeVirtResourceRequirements(ctx, userClient, config)
+		if err != nil {
+			return fmt.Errorf("error getting kubevirt resource requirements: %w", err)
+		}
+	case types.CloudProviderVsphere:
+		quotaReq, err = getVsphereResourceRequirements(config)
```
Comment (kron4eg) [inline pkg/ee/validation/machine/resourcequota.go:79]: I'd expect that vSphere might required customer CABundle too.

Comment (lsviben) [reply pkg/ee/validation/machine/resourcequota.go:79]: we probably would, but for vSphere we dont need to contact vSphere at all, we can just take the resource details from the machine spec

**Pull request merged**